Source/KakapoDB.swift

```diff
@@ -89,6 +127,34 @@ public class KakapoDB {
         return object
     }
     
+    public func update<T: Storable>(entity: T) -> Bool {
+        let index: Int? = barrierSync {
+            self.lookup(T).value.indexOf(entity)
+        }
+        
+        guard let indexToUpdate = index else { return false }
+        
+        barrierAsync {
```

Comment:
why don't we do everything in the `barrierSync` ? Should be deadlock safe. Can't the index become wrong if in the meanwhile something is changing the array (after barrier)? (hard to test)